browser/src/GenePage/GenePageContainer.tsx

```diff
@@ -8,8 +8,31 @@ import StatusMessage from '../StatusMessage'
 
 import GeneNotFound from './GeneNotFound'
 import GenePage from './GenePage'
+import {
+  heterozygousVariantCooccurrenceSeverities,
+  heterozygousVariantCooccurrenceAfCutoffs,
+  homozygousVariantCooccurrenceSeverities,
+  homozygousVariantCooccurrenceAfCutoffs,
+} from './VariantCooccurrenceCountsTable'
```

Comment:
Nothing necessarily actionable on this comment, so really just a comment, but having to import things from the particular component used within the gene page feels a bit odd.

I get why it's happening, that the container is where the query happens and the query is so unreasonably large that you map over those values to construct it instead of typing out however many lines, but it is a bit odd. 

I haven't looked as closely at the data, so I could very easily be missing why this doesn't work, but it seems as though there's an implementation where instead of however many distinctly named fields you manually ask for from GraphQL, arrays could be used with another field or two representing the cutoff and/or severity. That way the query could be comprehensible and wouldn't require programmatic creation.

Reply:
I'm thinking about this question more, and I don't really see what's so bad about big GraphQL queries. @mattsolo1 what was your concern about them?

Reply:
The debate here is whether this data is better represented in a "long" or "wide" format.

See: https://www.statology.org/long-vs-wide-data/

The wide format is easier to understand from a human-readable standpoint but more complex from a computer-readable standpoint because now we need to keep track of so many column names in the code.

Phil has chosen a very wide format, which means we need to name every column we want in the query. In fact, there are so many columns that we now need to programmatically generate the query.

So what is the actual use case for needing to query columns in this way? I think Phil previously said that the data is already "baked" by the pipeline, so it's ready to use in the app & and perhaps by API users who are interested in this data. This likely simplifies client-side code, since the work of shaping the data into the format you're looking for has already been done.  

However, since you're baking the data in way that is ready for UI display, you're in a sense of moving display logic into the pipeline step, which actually reduces the utility of the data for API users who may want to use it for some other purpose.

The other use case for Phil's implementation is maybe users want to query specific severity or cutoffs. GraphQL has a better solution for that: arguments (https://graphql.org/learn/queries/#arguments)

Riley said:

> there's an implementation where instead of however many distinctly named fields you manually ask for from GraphQL, arrays could be used with another field or two representing the cutoff and/or severity

Consider:

![image](https://user-images.githubusercontent.com/8867295/219018670-603a3943-407c-4419-b046-33f9e1459a3d.png)

Riley is suggesting a long table approach (i.e. moving the theoretically infinite number of "year" columns under one column called "year"). 

I agree, and I strongly suggest that we revise this PR as such. I know it's a lot of work, but I think it's worth it.

In the gnomAD codebase, the only other part of the GraphQL query that takes the long table approach is the GTEx query, which I think was a mistake, and now we're stuck with this long list of tissue names-as-fields that would be better represented under a variable called "tissue" 

https://github.com/broadinstitute/gnomad-browser/blob/a4c5891399036cf906269356fb96652fc8a891b4/browser/src/GenePage/GenePageContainer.tsx#L85

I'd rather we not repeat this mistake with the dual rare variants feature. I don't think it's purely aesthetic --  it will be much easier to understand and maintain this feature in the long term if we design it right. If we really need to get this feature out quickly for stakeholders let's go ahead and merge but I'd like to revise it as soon as we can.

Reply:
Yeah, I think I see where you're coming from now. My main concern with the long representation is that (for reasons that take a little explaining, I can if you really want the details) it ends up subverting type safety when you reshape it on client-side into the data structure we'll actually be using there.

Thing about that is, we didn't actually have type safety there anymore to speak of: the JSON encoding trick already blew that for us. While I never really like having to give that up, it's not as big a potential booby trap here as it might be otherwise, since we--this team specifically--control both ends of the transaction, the frontend browser code and the GraphQL server.

Thinking a little more, I think I do perhaps see a way to have both a long format and type safety on the frontend, will investigate.

Also more generally, as I think I remember you saying in the recent past @mattsolo1 , this does make me wonder if GraphQL is the best choice for our API--that's a conversation for another time but one worth having I'd say.